imap_processing/tests/lo/test_lo_science.py

```diff
@@ -266,3 +300,51 @@ def test_validate_parse_events(sample_data, attr_mgr):
 
     assert dataset["de_count"].values == 1998
     assert dataset["passes"].values == 8
+
+
+def test_organize_spin_data(fake_spin_data):
+    # Arrange
+    expected_dataset = xr.Dataset(
+        data_vars=dict(
+            num_completed=(["epoch"], np.array([0, 0])),
+            acq_start_sec=(["epoch"], np.array([0, 0])),
+            acq_start_subsec=(["epoch"], np.array([0, 0])),
+            acq_end_sec=(["epoch"], np.array([0, 0])),
+            acq_end_subsec=(["epoch"], np.array([0, 0])),
+            start_sec_spin=(
```

Comment:
Is "spin" here a coordinate, or should it be a data variable? I sort of assumed that these were a 1-1 mapping. A spin has an associated epoch, and epoch is already our coordinate here. So you'd want one or the other being coordinates, but not both, otherwise we have an (n x n) matrix that we index into as (i, i) instead of just i.

Reply:
Yes, the update I'm making to the data shape will use spin as the coordinate rather than epoch. Each epoch has 28 spins, but I'm going to flatten those out along one dimension called "spin" which will be an index coordinate for the number of spins in the pointing

Reply:
OK, now I think I might be understanding what you're trying to do now. One question though: Do you want each packet concatenated after each spin within that packet: `packet1-spin_field1, packet1-spin_field2, ... packet1-spin_field29, packet2-spin_field1, packet2-spin_field2, ...`?

Reply:
Correct. Based on my unit test it looks like that is correctly being concatenated.